### PR TITLE
feat: add flutter app Ory Network example 

### PR DESCRIPTION
**Pull request by sashatalalasha**
## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: flutter-ory-network/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+Create .env file with your project url in the root folder of the Flutter app
+
+```env
+ORY_BASE_URL=https://{your-project-slug}.projects.oryapis.com
+```
+
+### Run locally
+
+1. Install dependancies from `pubspec.yaml`
```
Comment (jonas-jonas) [inline flutter-ory-network/README.md]:
```suggestion
1. Install dependencies from `pubspec.yaml`
```

File: flutter-ory-network/lib/blocs/auth/auth_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            isLoading: false));
+      } else if (e case TwoFactorAuthRequiredException _) {
+        emit(state.copyWith(
+            isLoading: false, session: null, status: AuthStatus.aal2Requested));
+      } else if (e case UnknownException _) {
+        emit(state.copyWith(isLoading: false, errorMessage: e.message));
+      } else {
+        emit(state.copyWith(isLoading: false));
+      }
+    }
```
Comment (jonas-jonas) [inline flutter-ory-network/lib/blocs/auth/auth_bloc.dart]: I think these could be individual catch clauses instead, right?

File: flutter-ory-network/lib/main.dart [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  final options = BaseOptions(
+    baseUrl: baseUrl,
+    connectTimeout: const Duration(seconds: 10000),
+    receiveTimeout: const Duration(seconds: 5000),
+    headers: {
+      'Accept': 'application/json',
+    },
+    validateStatus: (status) {
+      // here we prevent the request from throwing an error when the status code is less than 500 (internal server error)
+      return status! < 500;
```
Comment (jonas-jonas) [inline flutter-ory-network/lib/main.dart]: What happens if, for whatever reason, the `status` is null here?

File: flutter-ory-network/lib/pages/login.dart [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+                    message.text,
+                    style: TextStyle(color: getMessageColor(message.type)),
+                  ),
+              // show logout button if aal2 requested, otherwise sign up
+              BlocSelector<AuthBloc, AuthState, bool>(
+                  bloc: (context).read<AuthBloc>(),
+                  selector: (AuthState state) =>
+                      state.status == AuthStatus.aal2Requested,
+                  builder: (BuildContext context, bool booleanState) {
+                    if (booleanState) {
```
Comment (jonas-jonas) [inline flutter-ory-network/lib/pages/login.dart]:
```suggestion
                  builder: (BuildContext context, bool loggedIn) {
                    if (loggedIn) {
```

**Review by jonas-jonas — COMMENTED** [review]
Code LGTM, apart from a few minor things. I think we can merge after they are addressed and then iterate over it as stuff changes.

File: flutter-ory-network/lib/main.dart (hunk 3 above)
Comment (sashatalalasha) [reply flutter-ory-network/lib/main.dart]: null exception will be thrown. I've updated the code

**Comment by vinckr** [comment]
I see the review has been adressed, is it ready to take another look?

**Review by jonas-jonas — APPROVED** [review]

**Pull request merged**